src/lib.rs

```diff
@@ -1058,9 +1096,12 @@ pub mod rpc {
     }
 }
 
+/// A request to a service. This can be either local or remote.
```

Comment:
maybe explain what the difference between remove and local is?

Reply:
It's a bit hard to put more explanation on this thing since it is fully generic intentionally. It is basically an Either with nicer case names than Ok / Err...

Added a bit more to the cases.